sam/exploration/tests/test_signalaligner.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+import numpy as np
+import pandas as pd
+
+from ..signalaligner import SignalAligner
+
+
+@pytest.mark.parametrize("N_aligned", [40, 50])
+@pytest.mark.parametrize("N1", [60, 90, 120])
+@pytest.mark.parametrize("N2", [60, 90, 120])
+@pytest.mark.parametrize("reference", [None, 0, 1])
+def test_signalaligner(N_aligned, N1, N2, reference):
```

Comment:
Nice, I actually prefer this to the unit test approach we use in the other tests